### PR TITLE
Nuri/feat:Blog homepage video banner

### PR DESCRIPTION
**Pull request by njazuli-deriv**
Changes:

-   Deriv Blog - Homepage video banner

## Type of change

-   [ ] Bug fix
-   [X] New feature
-   [ ] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: src/pages/blog/video-banner/carousel/_custom-carousel.js [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+    >
+        {color === 'black' ? (
+            <ChevronLeft black="true" />
+        ) : color === 'red' ? (
+            <ChevronLeft red="true" />
+        ) : color == 'custom' ? (
+            <ChevronLeft custom="true" />
+        ) : (
+            <ChevronLeft />
+        )}
```
Comment (sean-binary) [inline src/pages/blog/video-banner/carousel/_custom-carousel.js:43]: You can use a `switch` `case` here for better clarity.

File: src/pages/blog/video-banner/carousel/_custom-carousel.js [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+
+PrevButton.propTypes = {
+    chevron_style: PropTypes.object,
+    enabled: PropTypes.bool,
+    onClick: PropTypes.func,
+}
+
+export const NavigationButton = ({ color, is_enabled, onClick }) => (
+    <StyledDot onClick={onClick} color={is_enabled ? color : null} />
+)
```
Comment (sean-binary) [inline src/pages/blog/video-banner/carousel/_custom-carousel.js:55]: I'm assuming we can just use `StyledDot` for this case.

File: src/pages/blog/video-banner/carousel/_custom-carousel.js [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+    >
+        {color === 'black' ? (
+            <ChevronRight black="true" />
+        ) : color === 'red' ? (
+            <ChevronRight red="true" />
+        ) : color == 'custom' ? (
+            <ChevronRight custom="true" />
+        ) : (
+            <ChevronRight />
+        )}
```
Comment (sean-binary) [inline src/pages/blog/video-banner/carousel/_custom-carousel.js:78]: Same here.

File: src/pages/blog/video-banner/_VideoCarousel.js [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+    @media ${device.tabletL} {
+        width: 10px;
+        height: 10px;
+    }
+`
+
+const SeeMoreBtnMobile = styled(LinkButton)`
+    width: 100%;
+    max-width: 360px;
+    border: 1px solid #999999;
```
Comment (sean-binary) [inline src/pages/blog/video-banner/_VideoCarousel.js:116]:
```suggestion
    border: 1px solid var(--color-grey-5);
```

File: src/pages/blog/video-banner/_DBanner.js [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+    align-items: center;
+    cursor: pointer;
+`
+const PlayerIcon = styled.img`
+    width: 20px;
+    height: 20px;
+    @media ${device.tabletL} {
+        width: 20px;
+        height: 20px;
+    }
```
[the remark this answers is not in the document]
Comment (njazuli-deriv) [reply src/pages/blog/video-banner/_DBanner.js]: fixed in the latest commit

File: src/pages/blog/video-banner/_DBanner.js [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+        font-size: 16px;
+        line-height: 24px;
+        max-width: 100%;
+    }
+`
+const SmallDetailsWrapper = styled(Flex)`
+    height: 24px;
+    justify-content: flex-start;
+    align-items: center;
+`
```
[the remark this answers is not in the document]
Comment (njazuli-deriv) [reply src/pages/blog/video-banner/_DBanner.js]: fixed in the latest commit

File: src/pages/blog/video-banner/_DBanner.js [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+    const handleOpenVideo = (event) => {
+        if (event.defaultPrevented) return
+        setShow(true)
+    }
+
+    return (
+        <>
+            <ParentWrapper
+                bg_desktop={video_details[0].img_url_desktop}
+                bg_mobile={video_details[0].img_url_mobile}
```
[the remark this answers is not in the document]
Comment (njazuli-deriv) [reply src/pages/blog/video-banner/_DBanner.js]: fixed in the latest commit

File: src/pages/blog/video-banner/_VideoCarousel.js (hunk 4 above)
Comment (njazuli-deriv) [reply src/pages/blog/video-banner/_VideoCarousel.js]: fixed in the latest commit

File: src/components/hooks/use-browser-resize.js [hunk 8]
```diff
@@ -0,0 +1,26 @@
+import { useState, useEffect } from 'react'
+import { isBrowser } from 'common/utility'
+import { size } from 'themes/device'
+
+export const useBrowserResize = () => {
+    const [is_mobile, setMobile] = useState(false)
+    const [width, setWidth] = useState(isBrowser() ? window.innerWidth : 0)
```
Comment (sean-binary) [inline src/components/hooks/use-browser-resize.js:7]: This is great 👍 We're using this in a couple of places. We can replace them with this hook in the future.

File: src/pages/blog/video-banner/_DBanner.js [hunk 9]
```diff
@@ -0,0 +1,172 @@
...
+import PlayIcon from 'images/svg/blog/video/Triangle.svg'
+
+const ParentWrapper = styled(Flex)`
+    /* prettier-ignore */
+    background: ${(props) =>
+        props.bg_image
+            ? 'linear-gradient(251.14deg,rgba(14, 14, 14, 0.5632) 29.18%,rgba(7, 6, 6, 0.88) 85.14%),url(' +
+              props.bg_image +
+              ') no-repeat top left'
+            : 'linear-gradient(251.14deg, rgba(14, 14, 14, 0.5632) 29.18%, rgba(7, 6, 6, 0.88) 85.14%)'};
```
Comment (sean-binary) [inline src/pages/blog/video-banner/_DBanner.js:19]:
```suggestion
    background: ${(props) =>
        props.bg_image
            ? `linear-gradient(251.14deg,rgba(14, 14, 14, 0.5632) 29.18%,rgba(7, 6, 6, 0.88) 85.14%),url(${props.bg_image}) no-repeat top left`
            : 'linear-gradient(251.14deg, rgba(14, 14, 14, 0.5632) 29.18%, rgba(7, 6, 6, 0.88) 85.14%)'};
```

Comment (sean-binary) [reply src/pages/blog/video-banner/_DBanner.js:19]: We can use template strings here to make it a little more readable.

File: src/pages/blog/video-banner/_DBanner.js [hunk 10]
```diff
@@ -0,0 +1,172 @@
...
+    background-size: cover;
+    height: auto;
+    padding: 160px 0 80px;
+    max-width: 1600px;
+
+    @media ${device.tabletL} {
+        /* prettier-ignore */
+        background: ${(props) =>
+            props.bg_image
+                ? 'linear-gradient(251.14deg, rgba(14, 14, 14, 0.5632) 29.18%, rgba(7, 6, 6, 0.88) 85.14%),url(' +
```
Comment (sean-binary) [inline src/pages/blog/video-banner/_DBanner.js:33]: Same here.

File: src/pages/blog/video-banner/_DBanner.js [hunk 11]
```diff
@@ -0,0 +1,172 @@
...
+                : 'linear-gradient(251.14deg, rgba(14, 14, 14, 0.5632) 29.18%, rgba(7, 6, 6, 0.88) 85.14%)'};
+        background-size: cover;
+        padding: 73px 0 40px;
+    }
+`
+
+const BgWrapper = styled(Container)`
+    flex-direction: column;
+    justify-content: flex-start;
+`
```
Comment (sean-binary) [inline src/pages/blog/video-banner/_DBanner.js:45]: `Container` can take those as props by itself.

File: src/pages/blog/video-banner/_DBanner.js [hunk 12]
```diff
@@ -0,0 +1,172 @@
...
+const VideoDetailsWrapper = styled(Flex)`
+    height: auto;
+    flex-direction: column;
+    justify-content: flex-start;
+`
```
Comment (sean-binary) [inline src/pages/blog/video-banner/_DBanner.js:50]: Can take these as props as straight.

File: src/pages/blog/video-banner/_DBanner.js [hunk 13]
```diff
@@ -0,0 +1,172 @@
...
+    }
+`
+const TagWrapper = styled(Flex)`
+    width: auto;
+    padding: 1px 8px;
+    background: var(--color-orange-3);
+    border-radius: 8px;
+    align-items: center;
+`
+const Tag = styled(LinkText)`
```
Comment (sean-binary) [inline src/pages/blog/video-banner/_DBanner.js:83]: I've actually just used a regular header/text for this as there isn't gonna be a tags page in phase 1.

File: src/pages/blog/video-banner/_VideoCarousel.js [hunk 14]
```diff
@@ -0,0 +1,225 @@
...
+import PropTypes from 'prop-types'
+import VideoSrc from '../../markets/static/video/globe.mp4'
+import VideoPlayer from './_video-player'
+import { CustomCarousel } from './carousel/_custom-carousel'
+import { Flex } from 'components/containers'
+import { Header } from 'components/elements'
+import { LinkButton } from 'components/form'
+import device from 'themes/device'
+import { useBrowserResize } from 'components/hooks/use-browser-resize'
+import PlayIcon from 'images/svg/blog/video/Triangle.svg'
```
Comment (sean-binary) [inline src/pages/blog/video-banner/_VideoCarousel.js:12]: We gotta remember to use the same one once merged with my PR.

File: src/pages/blog/video-banner/_VideoCarousel.js [hunk 15]
```diff
@@ -0,0 +1,225 @@
...
+    justify-content: flex-start;
+`
+const DetailsWrapper = styled(Flex)`
+    flex-direction: column;
+    margin-left: 8px;
+    width: 180px;
+`
+const CarouselWrapper = styled(Flex)`
+    height: auto;
+`
```
Comment (sean-binary) [inline src/pages/blog/video-banner/_VideoCarousel.js:25]: Can be passed as props.

File: src/pages/blog/video-banner/_VideoCarousel.js [hunk 16]
```diff
@@ -0,0 +1,225 @@
...
+    max-width: 360px;
+    border: 1px solid var(--color-grey-5);
+    border-radius: 4px;
+    background-color: transparent;
+    padding: 6px 0;
+    margin: 24px auto 0;
+    height: auto;
+    align-items: center;
+`
+const StyledMoreBtnMobile = styled(Header)`
```
Comment (sean-binary) [inline src/pages/blog/video-banner/_VideoCarousel.js:113]: Are we able to use a `type` for this?

File: src/pages/blog/video-banner/_video-player.js [hunk 17]
```diff
@@ -0,0 +1,89 @@
...
+import { Flex } from 'components/containers'
+import CloseIcon from 'images/svg/close.svg'
+import device from 'themes/device'
+
+const VideoWrapper = styled.div`
+    position: fixed;
+    left: 0;
+    top: 0;
+    right: 0;
+    bottom: 0;
```
Comment (sean-binary) [inline src/pages/blog/video-banner/_video-player.js:13]:
```suggestion
    inset: 0;
```

Comment (sean-binary) [reply src/pages/blog/video-banner/_video-player.js:13]: You can use a shorter syntax for this 😄

File: src/pages/blog/video-banner/_video-player.js [hunk 18]
```diff
@@ -0,0 +1,89 @@
...
+    position: relative;
+    background-color: var(--color-black);
+    @media ${device.tablet} {
+        width: 90%;
+    }
+`
+
+const VideoPlayer = ({ video_src, closeVideo }) => {
+    const handleEscape = (e) => {
+        if (e.keyCode === 27) {
```
Comment (sean-binary) [inline src/pages/blog/video-banner/_video-player.js:46]:
```suggestion
        if (e.key === 'Escape') {
```

Comment (sean-binary) [reply src/pages/blog/video-banner/_video-player.js:46]: As Yashim pointed out, we should use the `event.key` as` event.keyCode` is marked as deprecated. Can refer to https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode

File: src/pages/blog/video-banner/carousel/_custom-carousel.js [hunk 19]
```diff
@@ -0,0 +1,282 @@
...
```
Comment (yashim-deriv) [inline src/pages/blog/video-banner/carousel/_custom-carousel.js:79]:
```suggestion
        <ChevronRight black={color === 'black'} red={color==='red'}  custom={color==='custom'} />
```

**Review by yashim-deriv — COMMENTED** [review]
Some suggestion for cleaner code

File: src/pages/blog/video-banner/_DBanner.js (hunk 9 above)
Comment (njazuli-deriv) [reply src/pages/blog/video-banner/_DBanner.js]: fixed in latest commit

File: src/pages/blog/video-banner/_DBanner.js (hunk 10 above)
Comment (njazuli-deriv) [reply src/pages/blog/video-banner/_DBanner.js]: fixed in latest commit

File: src/pages/blog/video-banner/_DBanner.js (hunk 11 above)
Comment (njazuli-deriv) [reply src/pages/blog/video-banner/_DBanner.js]: fixed in latest commit

File: src/pages/blog/video-banner/_DBanner.js (hunk 12 above)
Comment (njazuli-deriv) [reply src/pages/blog/video-banner/_DBanner.js]: fixed in latest commit

File: src/pages/blog/video-banner/_DBanner.js (hunk 13 above)
Comment (njazuli-deriv) [reply src/pages/blog/video-banner/_DBanner.js]: fixed in latest commit

File: src/pages/blog/video-banner/_VideoCarousel.js (hunk 14 above)
Comment (njazuli-deriv) [reply src/pages/blog/video-banner/_VideoCarousel.js:12]: Noted on this..also on color variables.

File: src/pages/blog/video-banner/_VideoCarousel.js (hunk 15 above)
Comment (njazuli-deriv) [reply src/pages/blog/video-banner/_VideoCarousel.js]: fixed in latest commit

File: src/pages/blog/video-banner/_VideoCarousel.js (hunk 16 above)
Comment (njazuli-deriv) [reply src/pages/blog/video-banner/_VideoCarousel.js]: fixed in latest commit

File: src/pages/blog/video-banner/_video-player.js (hunk 17 above)
Comment (njazuli-deriv) [reply src/pages/blog/video-banner/_video-player.js]: fixed in latest commit

File: src/pages/blog/video-banner/_video-player.js (hunk 18 above)
Comment (njazuli-deriv) [reply src/pages/blog/video-banner/_video-player.js]: fixed in latest commit

File: src/pages/blog/video-banner/carousel/_custom-carousel.js [hunk 20]
```diff
@@ -0,0 +1,282 @@
...
+                )}
+                {chevron_color && is_arrow && (
+                    <NextButton
+                        color={chevron_color}
+                        onClick={scrollNext}
+                        enabled={nextBtnEnabled}
+                        style={chevron_right}
+                        is_reviews={is_displayed_on_mobile}
+                    />
+                )}
```
[the remark this answers is not in the document]
Comment (njazuli-deriv) [reply src/pages/blog/video-banner/carousel/_custom-carousel.js]: fixed in latest commit

File: src/pages/blog/video-banner/carousel/_custom-carousel.js (hunk 19 above)
Comment (njazuli-deriv) [reply src/pages/blog/video-banner/carousel/_custom-carousel.js]: fixed in latest commit

File: src/pages/blog/video-banner/_VideoCarousel.js [hunk 21]
```diff
@@ -0,0 +1,210 @@
...
+                                    </Flex>
+                                </ItemsMainWrapper>
+                            )
+                        })}
+                    </CustomCarousel>
+                </Flex>
+                {is_mobile && (
+                    <SeeMoreBtnMobile to="/">
+                        <Header type="paragraph-2" color="white" align="center">
+                            See all videos
```
Comment (ShuvoHabib) [inline src/pages/blog/video-banner/_VideoCarousel.js:197]: What will happen, if someone click this button?

Comment (njazuli-deriv) [reply src/pages/blog/video-banner/_VideoCarousel.js:197]: It should bring the user to `All videos` page I believe..Correct @sean-binary ?

Comment (sean-binary) [reply src/pages/blog/video-banner/_VideoCarousel.js:197]: That's right. Should link to `/blog/videos`

**Review by sean-binary — DISMISSED** [review]

**Review by kevinw-binary — APPROVED** [review]

**Review by sean-binary — APPROVED** [review]

**Pull request merged**